File: Laboratory work 5 Cs/GeometricShapes.cs
using System;

namespace Laboratory_work_5_Cs
{
    abstract class Shape
    {
        public abstract double GetArea();
        public abstract double GetPerimeter();
    }

    class Triangle:Shape
    {
        public Triangle(double[,] coordinates)
        {
            if(coordinates.GetLength(dimension: 0) != 3 || coordinates.GetLength(dimension: 1) != 2)
                throw new Exception("This is not a triangle");
            this.coordinates = coordinates;
        }
        public override double GetArea()
        {
            return Math.Abs(0.5 * (coordinates[0, 0] * coordinates[0, 1] - coordinates[0, 1] * coordinates[1, 0] + coordinates[1, 0] * coordinates[2, 1] -
                                   coordinates[1, 1] * coordinates[2, 0] + coordinates[2, 0] * coordinates[0, 1] - coordinates[2, 1] * coordinates[0, 0]));
        }
        public override double GetPerimeter()
        {
            return Math.Sqrt(Math.Pow(coordinates[0, 0] - coordinates[1, 0], 2) + Math.Pow(coordinates[0, 1] - coordinates[1, 1], 2)) +
                   Math.Sqrt(Math.Pow(coordinates[1, 0] - coordinates[2, 0], 2) + Math.Pow(coordinates[1, 1] - coordinates[2, 1], 2)) +
                   Math.Sqrt(Math.Pow(coordinates[2, 0] - coordinates[0, 0], 2) + Math.Pow(coordinates[2, 1] - coordinates[0, 1], 2));
        }
        private double[,] coordinates;
    }

    class Circle : Shape
    {
        public Circle(double radius)
        {
            if (radius <= 0)
                throw new Exception("This is not a circle");
            this.radius = radius;
        }
        public override double GetArea()
        {
            return Math.Pow(radius, 2);
        }
        public override double GetPerimeter()
        {
            return 2 * Math.PI * radius;
        }
        private double radius;
    }
}

File: Laboratory work 5 C++/GeometricShapes.cpp
#include "GeometricShapes.h"
#include <iostream>
#include <math.h>

Triangle::Triangle(double coordinates[3][2])
{
    this->coordinates = new double* [3];
    for (int i = 0; i < 3; i++)
        this->coordinates[i] = new double[2];
    for (int i = 0; i < 3; i++)
        for (int j = 0; j < 2; j++)
            this->coordinates[i][j] = coordinates[i][j];
}
Triangle::~Triangle()
{
    for (int i = 0; i < 3; i++)
        delete[] coordinates[i];
    delete[] coordinates;
}
double Triangle::GetArea()
{
    return abs(0.5 * (coordinates[0][0] * coordinates[0][1] - coordinates[0][1] * coordinates[1][0] + coordinates[1][0] * coordinates[2][1] -
                      coordinates[1][1] * coordinates[2][0] + coordinates[2][0] * coordinates[0][1] - coordinates[2][1] * coordinates[0][0]));
}
double Triangle::GetPerimeter()
{
    return sqrt(pow(coordinates[0][0] - coordinates[1][0], 2) + pow(coordinates[0][1] - coordinates[1][1], 2)) +
           sqrt(pow(coordinates[1][0] - coordinates[2][0], 2) + pow(coordinates[1][1] - coordinates[2][1], 2)) +
           sqrt(pow(coordinates[2][0] - coordinates[0][0], 2) + pow(coordinates[2][1] - coordinates[0][1], 2));
}

Circle::Circle(double radius)
{
    if (radius <= 0)
        throw std::exception("This is not a circle");
    this->radius = radius;
}
double Circle::GetArea()
{
    return pow(radius, 2);
}
double Circle::GetPerimeter()
{
    return 2 * 3.1415926535897931 * radius;
}